state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -888,36 +747,39 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 					broadcast_id
 				);
 			} else {
-				Self::start_broadcast_attempt(BroadcastAttempt::<T, I> {
-					broadcast_attempt_id: next_broadcast_attempt_id,
-					..broadcast_attempt
-				});
+				Self::start_broadcast_attempt(broadcast_id, broadcast_attempt);
 			}
 		} else {
 			log::error!("No threshold signature data is available.");
 		};
 	}
 
-	fn start_broadcast_attempt(mut broadcast_attempt: BroadcastAttempt<T, I>) {
+	fn start_broadcast_attempt(
+		broadcast_id: BroadcastId,
+		mut broadcast_attempt: BroadcastAttempt<T, I>,
+	) {
 		T::TransactionBuilder::refresh_unsigned_data(&mut broadcast_attempt.transaction_payload);
 		TransactionMetadata::<T, I>::insert(
-			broadcast_attempt.broadcast_attempt_id.broadcast_id,
+			broadcast_id,
 			<<T::TargetChain as Chain>::TransactionMetadata>::extract_metadata(
 				&broadcast_attempt.transaction_payload,
 			),
 		);
 
-		let seed =
-			(broadcast_attempt.broadcast_attempt_id, broadcast_attempt.transaction_payload.clone())
-				.encode();
+		// Pass in the attempt count in as part of the seed to achieve pseudo-randomness.
+		let seed = (
+			broadcast_id,
+			Self::attempt_count(broadcast_id),
+			broadcast_attempt.transaction_payload.clone(),
+		)
+			.encode();
 		if let Some(nominated_signer) = T::BroadcastSignerNomination::nominate_broadcaster(
 			seed,
-			&FailedBroadcasters::<T, I>::get(broadcast_attempt.broadcast_attempt_id.broadcast_id)
-				.unwrap_or_default(),
+			&FailedBroadcasters::<T, I>::get(broadcast_id).into_iter().collect::<Vec<_>>(),
```

Comment:
can do the same as above, `nominate_broadcaster` could take an IntoIter in order to avoid this allocation, we call `iter()` on it inside this call anyway...

Reply:
Done